pkg/controller/config/config_controller.go

```diff
@@ -179,6 +184,18 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 		// if config is not found, we should remove cached data
 		if errors.IsNotFound(err) {
 			exists = false
+
```

Comment:
When this is part of opa.RemoveData(), and user adds a new kind to the config, it resets the metrics for rest of the kinds (except it adds it for the newly added kind), and they never gets populated

Reply:
If adding this after the call to `opa.RemoveData()` causes metrics not to recover, that is either a bug with the other parts of the metrics reporting or with syncing itself.

Because opa.RemoveData() removes __all__ data, we should reflect that, and if data doesn't repopulate, that is a genuine problem.

In this case it looks like it's a problem with the instrumentation, the `replayData` function in this file re-adds data to the cache, so it should be calling `addObject` as well.

Reply:
thanks, re-populating as part of `replayData` now